tokens.js

```diff
@@ -0,0 +1,33 @@
+require('dotenv').config()
+
+// https://stackoverflow.com/a/69058154/1413302
+const isTokenExpired = (token) => (Date.now() >= JSON.parse(Buffer.from(token.split('.')[1], 'base64').toString()).exp * 1000)
+
+/**
+ * Use the privately stored refresh token to generate a new access token for
+ * your RERUM-connected application. There is no way to authenticate this 
+ * process, so protect your refresh token and replace it if it is exposed. 
+ */
+function generateNewAccessToken() {
+    const payload = {
```

Comment:
```Java
            JSONObject tokenRequestParams = new JSONObject(); 
            tokenRequestParams.element("grant_type","refresh_token");
            tokenRequestParams.element("client_id",getRerumProperty("clientID"));
            tokenRequestParams.element("client_secret",getRerumProperty("rerumSecret"));
            tokenRequestParams.element("refresh_token",received.getString("refresh_token"));
            tokenRequestParams.element("redirect_uri", Constant.RERUM_BASE);
 ```
 You will need all of these parameters.  You only have `refresh_token` here, so don't forget.